agent/task_context.go

```diff
@@ -194,11 +229,12 @@ func (tc *taskContext) getExecTimeout() time.Duration {
 
 // commandBlock contains information for a block of commands.
 type commandBlock struct {
-	block       command.BlockType
-	commands    *model.YAMLCommandSet
-	timeoutKind timeoutType
-	getTimeout  func() time.Duration
-	canFailTask bool
+	block               command.BlockType
+	commands            *model.YAMLCommandSet
+	timeoutKind         timeoutType
+	getTimeout          func() time.Duration
```

Comment:
should we just embed heartbeatTimeoutOptions in here ?

Reply:
Hmm, we can't because 1. the exec timeout works in a different way and 2. timeout options are required for the regular (non-heartbeat) timeout mechanism, but are only conditionally required if the timeout ticker can time out the heartbeat as well.